src/target/riscv/riscv-013.c

```diff
@@ -3719,16 +3770,20 @@ read_memory_abstract(struct target *target, const riscv_mem_access_args_t args)
 {
 	assert(riscv_mem_access_is_read(args));
 
-	RISCV013_INFO(info);
-	bool use_aampostincrement = info->has_aampostincrement != YNM_NO;
-
 	memset(args.read_buffer, 0, args.count * args.size);
 
 	/* Convert the size (bytes) to width (bits) */
 	unsigned int width = args.size << 3;
 
-	/* Create the command (physical address, postincrement, read) */
-	uint32_t command = access_memory_command(target, false, width, use_aampostincrement, false);
+	uint32_t command = access_memory_command(target, /* virtual = */ false,
+			width, /* postincrement = */ true, /* is_write = */ false);
+	bool use_aampostincrement = !is_command_unsupported(target, command);
+	if (!use_aampostincrement)
+		/* It is already known that this abstract memory
+		 * access with aampostincrement=1 is not supported.
+		 * So try aampostincrement=0 right away. */
+		command = access_memory_command(target, /* virtual = */ false,
+				width, use_aampostincrement, /* is_write = */ false);
```

Comment:
It looks like the algorithm could be improved to also detect early unsupported memory read with aampostincrement=0.

No need to do it in this MR request, unless you want.

A simple TODO seems enough for now.

```suggestion
	if (!use_aampostincrement) {
		/* It is already known that this abstract memory
		 * access with aampostincrement=1 is not supported.
		 * So try aampostincrement=0 right away. */
		command = access_memory_command(target, /* virtual = */ false,
				width, /* postincrement = */ false, /* is_write = */ false);
		if (is_command_unsupported(target, command)) {
			LOG_TARGET_DEBUG(target, "Abstract memory read not supported (with or without aampostincrement)")
			return mem_access_result(...);
		 }
	}				
```

Reply:
Addressed

Reply:
@fk-sc, please, let's not do this in the same commit.
Current implementation returns different errors for the two consequtive runs of the same command in case abstract memory access is not supported:
`MEM_ACCESS_SKIPPED_ABSTRACT_ACCESS_CMDERR` vs `MEM_ACCESS_SKIPPED_AC_NOT_SUPPORTED`.

I believe this is confusing and should be addressed separately.
Please, add a `TODO` comment instead.

Reply:
I do not feel strongly. If @en-sc prefers to make this improvement in a separate PR, that is fine with me.

Reply:
Addressed